tests/src/item-namespace-staking.test.ts

```diff
@@ -0,0 +1,748 @@
+import * as anchor from "@project-serum/anchor";
+import {
+  Constants,
+  ItemProgram,
+  Idls,
+  NamespaceProgram,
+  StakingProgram,
+  Instructions,
+  Utils,
+  State,
+} from "@raindrops-protocol/raindrops";
+import { mintNFT, mintTokens } from "./utils/token";
+
+describe("Item-Namespace-Staking Flow", () => {
+  // Creates an anchor provider
+
+  // const RPC_URL = "http://localhost:8899";
+  const RPC_URL = "https://api.devnet.solana.com";
+  const walletKeypair = anchor.web3.Keypair.generate();
+
+  const provider = new anchor.AnchorProvider(
+    new anchor.web3.Connection(RPC_URL),
+    new anchor.Wallet(walletKeypair),
+    { commitment: "confirmed" }
+  );
+
+  // Sets required params
+
+  let itemProgram: ItemProgram;
+  let namespaceProgram: NamespaceProgram;
+  let stakingProgram: StakingProgram;
+
+  const componentScope = "none";
+  const itemClassMintKeypair = anchor.web3.Keypair.generate();
+  const itemClassIndex = 0;
+  const craftItemMintKeypair = anchor.web3.Keypair.generate();
+  const craftItemEscrowIndex = 0;
+  const itemMintKeypair = anchor.web3.Keypair.generate();
+  const itemIndex = 0;
+  const namespaceMintKeypair = anchor.web3.Keypair.generate();
+  const stakingMintKeypair = anchor.web3.Keypair.generate();
+  const stakingAmount = 100;
+  const stakingIndex = 0;
+  const stakingTransferAuthority = anchor.web3.Keypair.generate();
+
+  it("Airdrops sol to wallet", async () => {
```

Comment:
A bit overkill imo, we dont need to test solana rpc functions

Reply:
maybe all these are in sequence or something? and im not understanding the structure here.

Reply:
All tests are running in sequence. It's not for testing solana rpc functions. As you know, to send transactions, we need lamports in wallet.